testing/test_pathlib.py

```diff
@@ -89,3 +90,25 @@ def renamed_failed(*args):
     lock_path = get_lock_path(path)
     maybe_delete_a_numbered_dir(path)
     assert not lock_path.is_file()
+
+
+def test_long_path_during_cleanup(tmp_path):
+    """Ensure that deleting long path works (particularly on Windows (#6775))."""
+    path = tmp_path / ("a" * 250)
```

Comment:
what happens with you do "a" * 256?

Reply:
There is a limitation of the API that you can't create with any element longer than 260 without using long path API, which i was trying not to do in a test. I had this in an earlier version of the PR.